### PR TITLE
Uninitialised value fix

### PR DESCRIPTION
**Pull request by solvictor**
Prevent some conditional jumps on uninitialised values. Found with valgrind in the terrain generator and Client Handler.

### CODE REVIEW

**Comment by mjagdis** [comment]
Is that a real problem or just something that valgrind needs some help with? The Floor array are outputs from CalcFloorFrac as are how much of them are used. Any uninitialized part should be irrelevant.

If you really need them fully initialized it might be nicer to say, "int FloorX[MAX_SIZE] = { };" instead of using memset.

**Comment by solvictor** [comment]
Can someone help me figure out why Windows-x64-Debug is not passing ? Everything looks good.

**Comment by bearbin** [comment]
Not sure why it crashed, maybe the build agent got GCd or something. Reran and it worked :)

**Review by bearbin — CHANGES_REQUESTED** [review]
Looks like a reasonable change. Please could you add your name in the correct alpha-sorted position to the CONTRIBUTORS file to recognise your contribution under the correct license?

**Review by bearbin — COMMENTED** [review]
This looks sensible, I was just going to merge, but thinking again about what @mjagdis said, if we're actually getting uninitialised reads from these vars there's probably something wrong with the algorithm or its usage somewhere and we ought to look into that.

Do you have a reproducible test case that will trigger these uninitialised reads? I'd like to take a look and step through the code and debug what's actually happening - OFC feel free to do this yourself as well!

**Comment by bearbin** [comment]
I was referring to the noise change above, client handle change looks fine. If you could get a nice minimal example (maybe just make up a cpp file that uses the noise gen to trigger the read) that would be ideal!

**Comment by solvictor** [comment]
To reproduce the bug you can start a server with an empty world, the bug occurs when creating new chunks. To debug it you must compile in Debug mode and let the arrays uninitialized and launch the server with `valgrind --track-origins=yes ./Cuberite`. I don't know how the algorithm is supposed to work or what part of it is accessing theese values that are not supposed to be accessed.

**Comment by iamnoksio** [comment]
it's a perfect temp fix, push it

**Comment by mjagdis** [comment]
> it's a perfect temp fix, push it

The m_ProxyConnection bit, yes - it's a proper fix.

The Floor bit, no. There is no such thing as a "temp fix" and applying this would hide the only thing that suggests anything might be wrong here. So either someone needs to explain why there are accesses to Floor beyond a_Size and why those accesses should return 0 rather than some other value or someone needs to figure out what actually needs fixing.

I'd split it in two, a patch and an issue.

**Comment by mjagdis** [comment]
Forget the issue. See #5576

File: src/Noise/InterpolNoise.h [hunk 1]
```diff
@@ -348,13 +348,14 @@ class cInterpolNoise
...
-		int FloorY[MAX_SIZE];
+		int FloorX[MAX_SIZE] = {};
+		int FloorY[MAX_SIZE] = {};
 		NOISE_DATATYPE FracX[MAX_SIZE];
 		NOISE_DATATYPE FracY[MAX_SIZE];
 		int SameX[MAX_SIZE];
 		int SameY[MAX_SIZE];
 		int NumSameX, NumSameY;
+
 		CalcFloorFrac(a_SizeX, a_StartX, a_EndX, FloorX, FracX, SameX, NumSameX);
```
Comment (madmaxoft) [inline src/Noise/InterpolNoise.h:359]: These are not good changes. They hurt performance by needlessly filling the arrays with zeroes and then immediately overwriting the values with the proper calculated numbers.
Also the `CalcFloorFrac` call is tightly coupled to the declarations, the additional empty line breaks this logical tie.

As @mjagdis already pointed out, this needs a deeper fix, which he even provided, so please remove this from the PR.

**Review by madmaxoft — CHANGES_REQUESTED** [review]

File: src/Noise/InterpolNoise.h
Comment (madmaxoft) [inline src/Noise/InterpolNoise.h:1]: Changes to this file should be dropped from the PR altogether. Then it's fine to go.

**Review by madmaxoft — CHANGES_REQUESTED** [review]

**Review by madmaxoft — APPROVED** [review]
Awesome, thanks

**Comment by solvictor** [comment]
@bearbin Can you review please ?

**Comment by madmaxoft** [comment]
I took the liberty of merging it, the change itself is tiny so it's okay, and Bearbin's request was met.

**Comment by iamnoksio** [comment]
FINALLY DAMN, MERGING AFTER MONTHS IS CRAZY

**Pull request merged**